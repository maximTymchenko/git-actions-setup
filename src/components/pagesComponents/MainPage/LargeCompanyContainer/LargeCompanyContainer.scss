@import '../../../../styles/vars';

.large-company-container {
    position: relative;
    cursor: pointer;
    outline: 2px solid $secondary;
    box-shadow: 9px 9px 1px 1px rgba(57, 156, 203, 0.4);
    @include transition__outline;
    @include transition__shadow;
    @include background__position;
    display: grid;
    height: 360px;
    width: 100%;
    max-width: 655px;
    grid-template-columns: 1fr;

    @media screen and (max-width: $large__tablet) {
        max-width: 800px;
    }

    &--large {
        max-width: 800px;

        & .large-company-container__desc {
            margin-right: 270px;
            max-width: 500px;
        }
    }

    &--largest {
        max-width: 1080px;

        @media screen and (max-width: $large__tablet) {
            max-width: 800px;
        }
    }

    &--bg-main {
        background: url('../../../../assets/img/companies/services.jpeg');

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 0;
            }
        }

        &:hover::before {
            background-color: rgba(74, 85, 94, 0.8);
        }

        &::before {
            @include background__shadow;
            background-color: rgba(74, 85, 94, 0.7);
        }
    }

    &--bg-servers {
        background: url('../../../../assets/img/companies/servers.jpeg');
        padding-right: 50px;

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 5px;
            }
        }

        &:hover::before {
            background-color: rgba(74, 85, 94, 0.5);
        }

        &::before {
            @include background__shadow;

            background-color: rgba(74, 85, 94, 0.7);
        }
    }

    &--bg-rooms {
        background: url('../../../../assets/img/companies/office.jpeg');

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 0;
                margin-bottom: 0;
            }
        }

        &:hover::before {
            background-color: rgba(74, 85, 94, 0.5);
        }

        &::before {
            @include background__shadow;
            background-color: rgba(74, 85, 94, 0.3);
        }
    }

    &--bg-code {
        background: url('../../../../assets/img/companies/code.jpg');

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 0;
                margin-bottom: 0;
            }
        }

        &:hover::before {
            background-color: rgba(0, 24, 69, 0.6);
        }

        &::before {
            @include background__shadow;
            background-color: rgba(0, 24, 69, 0.5);
        }
    }

    &--bg-server-room {
        background: url('../../../../assets/img/companies/servers-room.jpg');

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 0;
                margin-bottom: 0;
            }
        }

        &:hover::before {
            background-color: rgba(0, 24, 69, 0.6);
        }

        &::before {
            @include background__shadow;
            background-color: rgba(0, 24, 69, 0.5);
        }
    }

    &--bg-agile {
        background: url('../../../../assets/img/companies/agile.jpeg');

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 0;
                margin-bottom: 0;
            }
        }

        &:hover::before {
            background-color: rgba(74, 85, 94, 0.5);
        }

        &::before {
            @include background__shadow;
            background-color: rgba(74, 85, 94, 0.3);
        }
    }

    &--bg-retail {
        background: url('../../../../assets/img/companies/retail.jpeg');

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 0;
                margin-bottom: 0;
            }
        }

        &:hover::before {
            background-color: rgba(0, 24, 69, 0.6);
        }

        &::before {
            @include background__shadow;
            background-color: rgba(0, 24, 69, 0.5);
        }
    }

    &--bg-ai {
        background: url('../../../../assets/img/companies/ai.jpeg');

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 0;
                margin-bottom: 0;
            }
        }

        &:hover::before {
            background-color: rgba(74, 85, 94, 0.5);
        }

        &::before {
            @include background__shadow;
            background-color: rgba(74, 85, 94, 0.3);
        }
    }

    &--bg-data {
        background: url('../../../../assets/img/companies/data.jpeg');

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 0;
                margin-bottom: 0;
            }
        }

        &:hover::before {
            background-color: rgba(0, 24, 69, 0.6);
        }

        &::before {
            @include background__shadow;
            background-color: rgba(0, 24, 69, 0.5);
        }
    }

    &--bg-desktop {
        background: url('../../../../assets/img/companies/desktop.jpg');

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 0;
                margin-bottom: 0;
            }
        }

        &:hover::before {
            background-color: rgba(74, 85, 94, 0.5);
        }

        &::before {
            @include background__shadow;
            background-color: rgba(74, 85, 94, 0.4);
        }
    }

    &--bg-phones {
        background: url('../../../../assets/img/companies/Phones.jpg');

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 0;
                margin-bottom: 0;
            }
        }

        &:hover::before {
            background-color: rgba(74, 85, 94, 0.6);
        }

        &::before {
            @include background__shadow;
            background-color: rgba(74, 85, 94, 0.5);
        }
    }

    &--bg-secondary {
        background-color: #3771c4;
    }

    &--bg-triangle {
        background-color: $secondary;
        position: relative;
        overflow: hidden;
        padding-bottom: 0;

        & .large-company-container__desc {
            @media screen and (max-width: $tablet) {
                align-self: end;
                padding-bottom: 5px;
            }
        }

        &::before {
            position: absolute;
            top: -155px;
            right: 0;
            content: '';
            display: block;
            width: 0;
            height: 0;
            border-top: 300px solid transparent;
            border-bottom: 300px solid transparent;
            border-right: 300px solid #b9008c;
        }
    }

    &:hover {
        outline: 4px solid $secondary;
        outline-offset: -2px;
    }

    &:active {
        @include button__press;
        box-shadow: 9px 9px 1px 1px rgba(57, 156, 203, 0);
    }

    &__logo {
        position: relative;
        z-index: 10;
        margin: 20px 0 70px 20px;

        width: 160px;
        position: relative;
        bottom: 30px;

        &--hide {
            display: none;
        }

        &--mb {
            margin-bottom: 117px;
        }

        &--mb-more {
            margin-bottom: 122px;
        }

        &--mb-less {
            margin-bottom: 102px;
        }

        @media screen and (max-width: $small__desktop) {
            width: 140px;
        }

        @media screen and (max-width: $large__tablet) {
            width: 120px;
        }

        @media screen and (max-width: $tablet) {
            margin: 2px 0 40px 10px;
            width: 100px;
        }
    }

    &__header {
        display: flex;
        justify-content: space-between;
        width: 100%;
        position: relative;
    }

    &__time {
        display: flex;
        flex-direction: column;
        position: absolute;
        right: 0;
        margin-top: 10px;
        margin-right: 10px;
    }

    &__date {
        opacity: 0.88;
        font-family: 'OktaNeue-Black';
        font-size: 10px;
        font-weight: 900;
        line-height: 1.1;
        letter-spacing: 0.5px;
        text-align: right;
        color: $background__primary;
        margin-bottom: 6px;
    }

    &__readtime {
        opacity: 0.88;
        font-family: 'OktaNeue-Bold';
        font-size: 14px;
        font-weight: bold;
        line-height: 1.71;
        text-align: right;
        color: $background__primary;
    }

    &__desc {
        position: relative;
        z-index: 10;
        margin-left: 10px;
        align-self: end;

        font-family: 'OktaNeue-Bold';
        font-size: 26px;
        font-weight: bold;
        line-height: 1.4;
        color: $background__primary;

        @media screen and (max-width: $small__desktop) {
            font-size: 22px;
            line-height: 1.2;
        }

        @media screen and (max-width: $small__tablet) {
            margin-left: 10px;
            font-size: 22px;
        }
    }

    // hashtags
    &__hashtags {
        margin-top: 15px;
        font-family: 'OktaNeue-Black';
        font-size: 11px;
        font-weight: 900;
        letter-spacing: 1px;
        color: $text__body;

        &--hide {
            display: none;
        }

        @media screen and (max-width: $mobile) {
            margin-bottom: 20px;
        }
    }
    &__label {
        background-color: $brand__primary;
        padding: 2px 3px;
        margin-right: 6px;
        font-family: 'OktaNeue-Black';
        font-size: 10px;
        font-weight: 900;
        line-height: 1.1;
        letter-spacing: 0.5px;
        color: $background__primary;
        text-transform: uppercase;

        &--editorial {
            background-color: $text__body;
        }

        &--whitepaper {
            color: $text__body;
            background-color: $color__confirmation;
        }
    }
}
