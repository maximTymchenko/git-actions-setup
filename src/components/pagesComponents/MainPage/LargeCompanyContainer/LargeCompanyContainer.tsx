import React, { FC } from 'react';
import './LargeCompanyContainer.scss';
import spectar from '../../../../assets/img/spectar.svg';

interface ILargeCompanyProps {
    img: any;
    description: string;
    label: string;
    hashtags?: string;
    hide?: string;
    typeBG?: string;
    size?: string;
}

const LargeCompanyContainer: FC = (props: ILargeCompanyProps) => {
    return (
        <div data-testid="large-company-container" className='large-company-container__wrapper'>
            <div className={`large-company-container large-company-container--${props.typeBG} large-company-container--${props.size}`}>
                <img className="large-company-container__logo" src={props.img} alt="Spectar" />
                <p className="large-company-container__desc">{props.description}</p>
            </div>
            <p className={`large-company-container__hashtags large-company-container__hashtags--${props.hide}`}
            >>
                <span className="large-company-container__label">{props.label}</span>
                {props.hashtags}
            </p>
        </div>
    );
};

export default LargeCompanyContainer;
