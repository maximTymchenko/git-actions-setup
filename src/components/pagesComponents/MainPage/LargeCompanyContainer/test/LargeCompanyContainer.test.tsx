import React from 'react'
import LargeCompanyContainer from '../LargeCompanyContainer'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import walmart from '../../../../../assets/img/walmart.svg'

it('renders elem correctly', () => {
  const { getByTestId } = render(
    <LargeCompanyContainer
      img={walmart}
      description="Increasing Walmart weekly ads conversion with a custom web app for Ad Tech
        company"
      label="CASE STUDY"
      hashtags="Amazon, Retail & E-commerce, React, Web development, +13 more"
    />
  )
  expect(getByTestId('large-company-container')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer
    .create(
      <LargeCompanyContainer
        img={walmart}
        description="Increasing Walmart weekly ads conversion with a custom web app for Ad Tech
        company"
        label="CASE STUDY"
        hashtags="Amazon, Retail & E-commerce, React, Web development, +13 more"
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
