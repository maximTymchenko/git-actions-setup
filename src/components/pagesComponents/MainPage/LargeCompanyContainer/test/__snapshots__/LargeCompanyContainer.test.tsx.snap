// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="large-company-container__wrapper"
  data-testid="large-company-container"
>
  <div
    className="large-company-container large-company-container--undefined large-company-container--undefined"
  >
    <img
      alt="Spectar"
      className="large-company-container__logo"
      src="test-file-stub"
    />
    <p
      className="large-company-container__desc"
    >
      Increasing Walmart weekly ads conversion with a custom web app for Ad Tech company
    </p>
  </div>
  <p
    className="large-company-container__hashtags large-company-container__hashtags--undefined"
  >
    &gt;
    <span
      className="large-company-container__label"
    >
      CASE STUDY
    </span>
    Amazon, Retail & E-commerce, React, Web development, +13 more
  </p>
</div>
`;
